#!/usr/bin/env python
PACKAGE = "ping_pong_follower"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#Area
gen.add("filterByArea",     bool_t,   0, "filterByArea",  True)
gen.add("minArea",          int_t,    0, "minArea", 50,  0, 50*50)

#Circularity
gen.add("filterByCircularity",   bool_t,    0, "filter By Circularityr",  True)
gen.add("minCircularity",       double_t,    0, "min Circularity", 0.1,  0, 1)

#Convexity
gen.add("filterByConvexity",   bool_t,   0, "filter By Convexity",  True)
gen.add("minConvexity",    double_t,    0, "min Convexity", 0.87,  0, 1)

#Inertia
gen.add("filterByInertia",   bool_t,   0, "filter By Inertia",  True)
gen.add("minInertiaRatio",    double_t,    0, "minInertiaRatio", 0.01,  0, 100)

exit(gen.generate(PACKAGE, "ping_pong_follower", "Blobber"))
